### PR TITLE
Add npm registry setting

### PR DESCRIPTION
**Pull request by ele828**
Implement issue #42.

image
This is comparison of time spent in downloading Gifted Messenger component via NPM official registry and Taobao CDN registry.

image
I just temporarily added this setting to ```Editor``` tab, but I think it's better to add it to ```General``` tab or another tabs.

### CODE REVIEW

File: web/src/scripts/containers/TabbedEditor.jsx [hunk 1]
```diff
@@ -302,6 +306,7 @@ const mapStateToProps = (state, ownProps) => {
       showInvisibles: state.preferences[CATEGORIES.EDITOR][PREFERENCES.EDITOR.SHOW_INVISIBLES],
       styleActiveLine: state.preferences[CATEGORIES.EDITOR][PREFERENCES.EDITOR.HIGHLIGHT_ACTIVE_LINE],
       showIndentGuides: state.preferences[CATEGORIES.EDITOR][PREFERENCES.EDITOR.SHOW_INDENT_GUIDES],
+      npmRegistry: state.preferences[CATEGORIES.EDITOR][PREFERENCES.EDITOR.NPM_REGISTRY],
```
Comment (dabbott) [inline web/src/scripts/containers/TabbedEditor.jsx]: This `options` object is specifically for CodeMirror text editor options. Can you move `npmRegistry` a few lines up into the parent object `return {...}`?

File: desktop/src/handlers/moduleHandler.js [hunk 2]
```diff
@@ -177,7 +177,14 @@ class ModuleHandler {
         bridge.send(startProgressBar(name, 0))
 
         try {
-          npm.run(['install', '-S', `${name}@${version}`], {cwd: installPath}, (err) => {
+          const command = [
+            'install', '-S', `${name}@${version}`,
+            ...options.registry && ['--registry', options.registry]
```
Comment (dabbott) [inline desktop/src/handlers/moduleHandler.js]: What does `...options.registry` do here? Should it just be `options.registry` without the `...`?

**Comment by dabbott** [comment]
Other than those 2 little things, looks great.

Yeah, I agree, it should go in a different tab. Let's leave it here for now though. I'm probably going to add a `Components` settings tab soon, so I will move it there when I do.

File: desktop/src/handlers/moduleHandler.js (hunk 2 above)
Comment (ele828) [inline desktop/src/handlers/moduleHandler.js]: Here, it equals to

```javascript
const command = ['install', '-S', `${name}@${version}`]
if (options.registry) {
   command.concat(['--registry', options.registry])
}
```

**Pull request merged**